### PR TITLE
SJSU-1015, Implement Booking/Payment Page

### PR DESCRIPTION
**Pull request by SandroSal**
**what I changed**
- linked BookingInfo.js and Payment.js to PaymentPage.js, so it can collect all data and forward it to backend
- added rewards points and discount functionality
- redesigned page (especially for the mobile screen sized)
- info button for rewards

**for Testing**
- you can put 'console.log(req.body);' into 'bookin.js'
- you can also play around with 'discounts' in 'indivHotel.js'
- and with 'rewardsPoints' in'BookingInfo.js'

### CODE REVIEW

File: front_end/src/components/payment/BookingInfo.js [hunk 1]
```diff
@@ -1,62 +1,262 @@
...
+    display: "none"
+  }
+});
+
+class BookingInfo extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      open: false,
+      checkedA: false,
```
Comment (ynoe) [inline front_end/src/components/payment/BookingInfo.js]: What does checkedA mean?

Comment (ynoe) [reply front_end/src/components/payment/BookingInfo.js]: ok availability...

File: front_end/src/components/payment/BookingInfo.js [hunk 2]
```diff
@@ -1,62 +1,262 @@
...
+            actualRewardsDiscount: 0
+          },
+          () => {
+            var rewardsPointsEarned =
+              100 *
+              (this.props.bookingData.tempBookingData.subtotal +
+                this.props.bookingData.tempBookingData.subtotal * 0.1 -
+                this.props.bookingData.tempBookingData.discounts -
+                this.state.actualRewardsDiscount);
+
```
Comment (ynoe) [inline front_end/src/components/payment/BookingInfo.js]: O.O

File: front_end/src/components/payment/BookingInfo.js (hunk 1 above)
Comment (SandroSal) [reply front_end/src/components/payment/BookingInfo.js]: I can rename it. Its actually the checkbox for applying rewards points :D

**Pull request merged**